mods/cnc/maps/gdi10a/gdi10a.lua

```diff
@@ -0,0 +1,413 @@
+	BaseBlueprints = {}
+	for i = 1, #RebuildableStructs do
+		local structure = RebuildableStructs[i]
+		local blueprint = {
+			Type = structure.Type,
+			Location = structure.Location,
+		}
+		table.insert(BaseBlueprints, blueprint)
+		SetupNodBuilding(blueprint, structure, false)
+	end
+end
+
+SetupNodBuilding = function(blueprint, structure, autoRepair)
+	Media.Debug("Setup " .. tostring(structure) .. " as " .. ActorString(blueprint) .. ' Bal$' .. BankBalance(Nod))
+	Trigger.OnKilled(structure, function()
+		-- Add to build queue
+		Media.Debug(string.format('Bldg killed: Another in progress? %s  Queue_len=%d',
+			tostring(rebuildingInProgress), #CyardBuildQueue))
+		table.insert(CyardBuildQueue, blueprint)
+		if not rebuildingInProgress then
+			-- Build queue was empty; start building now
+			-- This ensures we only build one structure at a time
+			ProcessBuildQueue(CyardBuildQueue)
+		end
+	end)
+	if autoRepair then
+		RepairBuilding(Nod, structure, 0.75)
+	end
+	-- NOTE If newly built, it enters the world on the next tick
+	if structure.Type == 'hand' or structure.Type == 'pyle' then
+		Trigger.AfterDelay(AttackCooldown[Difficulty], function()
+			print('Start infantry production+')
+			ProduceUnit(structure, InfProducer)
+			print('Start infantry production-')
+		end)
+	elseif structure.Type == 'afld' or structure.Type == 'weap' then
+		Trigger.AfterDelay(AttackCooldown[Difficulty], function()
+			print('Start tank production+')
+			ProduceUnit(structure, ArmorProducer)
+			print('Start tank production-')
+		end)
+	end
+	-- New units' first move
+	if structure.Type == 'hand' or structure.Type == 'afld' then
+		structure.RallyPoint = waypoint11.Location
+	end
+end
+
+-- Replace structures when destroyed
+ProcessBuildQueue = function(queue)
+	if rebuildingInProgress then
+		local s='ProcessBuildQueue: should not happen while build in progress! Queue_len='..#queue
+		Media.Debug(s)
+		print(s)
+	end
+	while #queue > 0 do
+		local rc = RebuildFromBlueprint(queue)
+		Media.Debug('RebuildFromBlueprint: ' .. rc)
+		if rc == 'cancelled' then
+			table.remove(queue, 1)
+		elseif rc == 'insufficient_funds' then
+			-- Continue processing later
+			Trigger.AfterDelay(DateTime.Seconds(5), function()
+				ProcessBuildQueue(queue)
+			end)
+			break
+		elseif rc == 'in_progress' then
+			-- Stop processing the queue while it builds
+			table.remove(queue, 1)
+			break
+		else
+			Media.Debug("Unknown return from RebuildFromBlueprint: " .. tostring(rc))
+		end
+	end
+end
+
+-- Helper for ProcessBuildQueue
+-- Return true when the item is finished and we should move to the next item in queue
+-- Return false when the item is in progress and we should not move to the next item the queue
+RebuildFromBlueprint = function(queue)
+	if #queue == 0 then
+		s = 'RebuildFromBlueprint: queue empty!'
+		Media.Debug(s)
+		print(s)
+		return 'cancelled'
+	end
+	local blueprint = queue[1]
+	local cyard = GetNodCyard()
```

Comment:
```suggestion
	local cyard = Nod.HasPrerequisites({ "fact" })
```
The global `CheckForBase` from `campaign.lua` could also stand in for `GetNodCyard` if you'd like. Either way, it seems a bit redundant.

Reply:
`CheckForBase` counts the number of buildings that match the specified types.  As does `HasPrerequisites`.

But I need a handle to the building itself (if it is not lost).  What is redundant about my code?  In TD you need the `cyard` object to build buildings.

Reply:
I didn't notice the use of the actor further down in the function. Please pardon my derp.